checker/src/org/checkerframework/checker/lock/LockPropagationTreeAnnotator.java

```diff
@@ -0,0 +1,37 @@
+package org.checkerframework.checker.lock;
+
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.util.Elements;
+
+import org.checkerframework.checker.lock.qual.GuardedBy;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.treeannotator.PropagationTreeAnnotator;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import com.sun.source.tree.BinaryTree;
+
+public class LockPropagationTreeAnnotator extends PropagationTreeAnnotator {
+
+    /** Annotation constants */
+    protected final AnnotationMirror GUARDEDBY;
+
+    public LockPropagationTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
+        super(atypeFactory);
+
+        Elements elements = atypeFactory.getElementUtils();
+        GUARDEDBY = AnnotationUtils.fromClass(elements, GuardedBy.class);
+    }
+
+    @Override
+    public Void visitBinary(BinaryTree node, AnnotatedTypeMirror type) {
+        if (((LockAnnotatedTypeFactory)atypeFactory).isBinaryComparisonOperator(node.getKind())) {
+            // TODO: Is any defaulting done by super.visitBinary needed here?
+            type.replaceAnnotation(GUARDEDBY);
```

Comment:
This doesn't seem to be doing any "propagation".  So why is it in a class called LockPropagationTreeAnnotator? Instead, this class should extend TreeAnnotator and createTreeAnnotator should list it before PropagationTreeAnnotator.